File: matrixSearch.java

// Time Complexity :O(n+m)
// Space Complexity :O(1)
// Did this code successfully run on Leetcode :Yes
// Three line explanation of solution in plain english

start from the bottom left corner if target is greater than move to next column else move to the previous row. 

// Your code here along with comments explaining your approach
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int row = matrix.length -1;
        int col = 0;
        while(row >= 0 && col <matrix[0].length)
        {
            if(matrix[row][col] > target)
            {
                row--;
            }
            else if(matrix[row][col] < target)
            {
                col++;
            }
            else
            {
                return true;
            }
        }
        return false;
    }
}

File: matrix_search.py
# Time Complexity :
# O(N+M)  - Size of the Matrix

# Space Complexity :
# O(1) - There is no auxillary space being allocated

# Did this code successfully run on Leetcode :
#Yes

#We start from one corner. If our current value is larger than the target, then we move towards the smaller value in the matrix
#If the current value is saller than the target, then we move towarsd the larger value in the matrix
#Since the rows and columns are sorted, we always know which direction to move in

class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        i = 0 
        j = len(matrix[0]) -1

        while (i < len(matrix) and j >= 0 ):
            if matrix[i][j] == target :
                return True

            if matrix[i][j] < target:
                i += 1
            else :
                j -= 1
